math/mode.go

```diff
@@ -0,0 +1,61 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+//package constraints
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+type Number interface {
+	constraints.Float | constraints.Integer
+}
+
+// Error when array is empty
+var emptyModeArr = errors.New("Array is empty")
+
+func Mode[T Number](numbers []T) (T, error) {
+
+	// Indicate numbers of checking(looping)
+	n := len(numbers)
+
+	// Assign when there is a Mode Value
+	maxCount := 0
+	var maxValue T = 0
+
+	if n == 0 {
+		return 0, emptyModeArr
+	} else {
+
+		// Check for each value has any repeated value
```

Comment:
Please remove all comment explaining the code.

The code should be self explanatory, which it is in this case. Use comments only when you're explaining **"why"** you're doing something not **"what"** you're doing.